File: ReferenceWRLImplementation/CSharpUnitTestProject/ActivatableStaticClassUnitTests.cs
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
using ReferenceWRLImplementation;

namespace CSharpUnitTestProject
{
    [TestClass]
    public class ActivatableStaticClassUnitTests
    {
        [TestMethod]
        public void StaticMethodTest()
        {
            uint expectedVal = 30;
            var actualVal = ActivatableStaticClass.StaticMethod();
            Assert.AreEqual(expectedVal, actualVal);
        }

        [TestMethod]
        public void NonStaticMethodTest()
        {
            uint expectedVal = 35;
            var activatableStaticClass = new ActivatableStaticClass();
            var actualVal = activatableStaticClass.NonStaticMethod();
            Assert.AreEqual(expectedVal, actualVal);
        }
    }
}

File: ReferenceWRLImplementation/CXUnitTestApp/ActivatableStaticClassUnitTests.cpp
#include "pch.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace ReferenceWRLImplementation;

namespace CXUnitTestApp
{
    TEST_CLASS(ActivatableStaticClassCPPUnitTests)
    {
    public:
        TEST_METHOD(StaticMethodCPPTest)
        {
            UINT32 expectedVal = 30;
            auto actualVal = ActivatableStaticClass::StaticMethod();
            Assert::AreEqual(expectedVal, actualVal);
        }

        TEST_METHOD(NonStaticMethodCPPTest)
        {
            UINT32 expectedVal = 35;
            auto activatableStaticClass = ref new ActivatableStaticClass();
            auto actualVal = activatableStaticClass->NonStaticMethod();
            Assert::AreEqual(expectedVal, actualVal);
        }
    };
}